### DISCUSSION TITLE
Cannot compile C files during the EGSnrc configuration?

### DISCUSSION THREAD

**Discussion by jcampbell555** [nrc-cnrc/EGSnrc#748] [Questions] [answered]
Getting an error that I can't seem to find anything for on this forum. Downloaded all the necessary software and I'm at the last step of installing EGS but I got this message:

```
GNU Fortran (x86_64-posix-seh-rev1, Built by MinGW-W64 project) 6.3.0


 Using xml test file C:\Users\Josh C\EGSnrc\HEN_HOUSE\pieces\tests_win.xml

 ===>   Checking for availability of system dependent functions 


Can the compiler produce object files? .................... yes
Can the compiler produce executables? ..................... yes
Does "exit" or "exit_" work properly? ..................... yes [exit]
 Compiling "egs_exit.c" since "exit" and "stop" failed ?... yes
Does the "system" function return an exit status ? ........ yes
Is "flush" or "flush_" available? ......................... yes [flush]
Is the "date_and_time" subroutine available ? ............. yes
Is "fdate" or "fdate_" subroutine available? .............. yes [fdate]
Is "secnds" or "secnds_" function available? .............. yes [secnds]
Is "date" or "date_" subroutine available? ................ no 
Is "time" or "time_" subroutine available? ................ no 
Is "etime" or "etime_" function available? ................ yes [etime]
Is function "lnblnk" available? ........................... yes
Does "hostnm" or "hostnm_" work properly? ................. yes [hostnm]
Is "lstat" available and does it work properly? ........... no 
Does "inquire" work properly? ............................. yes
 Compiling a C program to find out the file size ?......... yes
Is record length known? ................................... yes [4]
Is "integer*8" available? ................................. yes [integer*8]
Is "integer*2" available? ................................. yes [integer*2]
Can endianess of machine be determined?.................... yes [little endian]
 Compiling a C program to find out the name mangling scheme ? yes
Is the name mangling scheme known ?........................ yes [lower case, _ and _]

************              Tests concluded !          *******


===> Creating C Utilities for EGSnrc...


Building the utilities object file ... 

Could egs_c_utils.c be compiled? .......................... no 
Could load_beamlib.c be compiled? ......................... no 
Could load_vculib.c be compiled? .......................... no 
Could read_write_pardose.c be compiled? ................... no 

Failed to compile egs_c_utils.c. This may be due to a variety of reasons
(e.g. non-standard include files, unsupported file lock operations, etc.
Will disable now the availability of a C compiler. 
If you want to try to make the new parallel processing implementation in 
cavrznrc and dosrznrc work, 
  1. Go to $machine_mortran_dir 
  2. Edit egs_c_utils.h and egs_c_utils.c to make them compile with your 
    C compiler 
 3. Put 
     REPLACE {$HAVE_C_COMPILER(#);} WITH {{EMIT;{P1}define HAVE_C_COMPILER};} 
    in machine.macros 
 4. Put  
    $HAVE_C_COMPILER(#); 
    near the beginning of your user code.
  5. If your compiler requires some unusual options, include directives, etc., 
     change the variable C_FLAGS in the spec file $conf_file in the 
     $HEN_HOUSE/specs/ directory. 


 **** Failed to compile load_vculib.c!!! ****
 This is the tool for using sources 20 and 21 in BEAMnrc.

Failed compiling read_write_pardose.c!
This is required to compile DOSXYZnrc!


Testing egs_c_utils skipped due to previous errors...


Assuming there is no need to pass a flag
to the compiler for opening a DSO on Windows

using Win API function LoadLibrary().


===> Creating configuration file ...



===> Creating machine.macros ...



 *** System file C:\Users\Josh C\EGSnrc\HEN_HOUSE\lib\win3264\machine.macros successfully created ***



===> Creating machine.f ...



===> Creating machine.mortran ...



 *** System file C:\Users\Josh C\EGSnrc\HEN_HOUSE\lib\win3264\machine.mortran successfully created ***


===> Creating dosxyznrc_win3264.spec ...
 failed 

Failed creating dosxyz_win3264.spec
file. You will not be able to use 
DOSXYZnrc for in-phantom dose calculations.

===> Appending VCU library to dosxyznrc_win3264.spec ...

===> Compiling Mortran3 ...


Makefile:31: C:\Users\Josh: No such file or directory

Makefile:31: C\EGSnrc\HEN_HOUSE\specs\win3264.conf: No such file or directory

mingw32-make: *** No rule to make target 'C\EGSnrc\HEN_HOUSE\specs\win3264.conf'.  Stop.

Compilation failed!

===> Compiling Pegs4 ...


Makefile:31: C:\Users\Josh: No such file or directory
Makefile:31: C\EGSnrc\HEN_HOUSE\specs\win3264.conf: No such file or directory

mingw32-make: *** No rule to make target 'C\EGSnrc\HEN_HOUSE\specs\win3264.conf'.  Stop.

Compilation failed!

****************************************************
* Failed building mortran3, the string preprocessor*
* utility for generating Fortran code. You won't be*
* able to compile any Mortran source code therefore*
* your system will not be functional.              *
****************************************************




***** EGSnrc installation FAILED, see log file for details  *****


*********************************
*     Elapsed time: 41.148 s *
*********************************
```

Any help would be greatly appreciated. Thanks!

**Comment by ftessier** [comment]
It appears that the EGSnrc configuration tool is not able to compile C files on your system. This could be either because there is no C compiler installed on your system, or because it is not visible on the system's PATH from the point of view of the configuration tool. It could also be due to an incompatibility with the compiler version 6.3.0. You can check this by:

```
  1. Go to $machine_mortran_dir
  2. Edit egs_c_utils.h and egs_c_utils.c to make them compile with your 
    C compiler 
```

Normally there is a configuration log file which may contain additional information about the failure point. Please share that here if possible to diagnose further.

**Reply by thomashw28** [reply]
what should i edit in that egs_c_utils.h and egs_c_utils.c complex code?

**Reply by rtownson** [reply]
@thomashw28 please make a new discussion that includes as many details about the issue you're encountering as possible. Generally there is no need to touch `egs_c_utils`.

**Reply by ROUZII3** [reply]
I'm encountering difficulties compiling egs_c_utils.c and egs_c_utils.h, despite following the provided instructions and attempting to adjust the GCC version. I'm seeking assistance to resolve this issue as it persists despite my efforts. Any guidance or suggestions would be greatly appreciated.
[configure.log](https://github.com/nrc-cnrc/EGSnrc/files/14471513/configure.log)
[the error msg.txt](https://github.com/nrc-cnrc/EGSnrc/files/14471519/the.error.msg.txt)

**Reply by rtownson** [reply]
@ROUZII3 next time please make a new discussion and just refer to this one if you think it's relevant. 

In your log I'm seeing /c/, why is it using that instead of C:/? Why does it look for make in "/c/Program Files (x86)/GnuWin32/bin/make", when you also have mingw installed? It should be using the mingw version of make. Either use entirely GnuWin32 (with which I'm not familiar), or not at all.

**Reply by ROUZII3** [reply]
I apologize for the interruption in this discussion. I noticed a similarity between my issue and the one being discussed here. Would it be appropriate for me to start a new discussion about my specific problem?
I believe the '/c/' issue may be due to using Git Bash on Windows, where it should be translated to 'C:'. I'm attempting to install EGS on Windows and have uninstalled GNUwin32. Despite reinstalling EGS configuration, I'm encountering the same issue. As a newcomer to EGS, I'm uncertain about the next steps. I truly appreciate your time and assistance in helping me resolve this matter.

**Reply by rtownson** [reply]
Are you installing using the installer GUI? Don't use the command line configuration tool on windows. You should do the installation without using git bash at all, following the installation instructions. Check that your environment variables don't include GNUwin32, and in the installer only use paths like C:/. Try this again, and if it doesn't working, you're welcome to post your new installation log with errors in a new discussion.

**Reply by ROUZII3** [reply]
I'm utilizing the command line configuration tool as per the instructions in the EGS_brachy manual, as my primary objective is to conduct simulations in brachytherapy. However, I've encountered an obstacle while attempting to configure EGSnrc using the provided configuration shell script, resulting in compilation issues. Given this situation, would it be advisable for me to initiate a new discussion thread in the EGSnrc or EGS_brachy ?
Additionally, I want to express my gratitude for the helpful responses I've received so far. Thank you for your assistance.

**Reply by rtownson** [reply]
@ROUZII3  Those egs_brachy installation instructions are intended for linux. Though they should mostly work using git bash, except for the `./HEN_HOUSE/scripts/configure` step. Replace that step with installing EGSnrc using the GUI installer for Windows. Make sure to check the checkboxes in the installer to place GUI shortcuts and set environment variables.

If that doesn't work, clean up everything and start again using the EGSnrc Windows installation instructions. Then follow the egs_brachy instructions for existing EGSnrc installations.

Yes, if you continue to have issues, please post a new discussion here.

**Comment by rtownson** [comment, marked as answer]
There is a space in the EGSnrc path, and this is not allowed: `C:\Users\Josh C\EGSnrc`

If possible, I also recommend the shortest path possible, since there are still some path length restrictions in parts of the code. E.g. `C:\EGSnrc`

**Reply by ftessier** [reply]
Thanks, again, I missed it... 😩

**Reply by jcampbell555** [reply]
Thanks for the help. I ended up accidentally resetting my entire windows interface trying to change my users' folder name to get rid of the space. It seems to be working though. Much appreciated!
